scripts/protoc_installer.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright 2024 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#set -eu -o pipefail
+
+# Perform installation of protoc from source based on OS.
+
+PROTOC_VERSION="25.2"
+
+# Function to download pre-built binaries for Linux with
+# ARCH as $1, OS as $2, and WORKDIR as $3 arguments.
+download_binary() {
+  # Download URL (adjust if a newer release is available)
+  DOWNLOAD_URL="https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-$2-$1.zip"
+  # Download and unzip
+  curl -LO "$DOWNLOAD_URL"
+  INSTALL_DIR="${3:-${GOBIN:-${GOPATH:-$HOME/go}/bin}}"
+  unzip "protoc-${PROTOC_VERSION}-$2-$1.zip" -d $INSTALL_DIR
+  rm "protoc-${PROTOC_VERSION}-$2-$1.zip"
+}
+
+# Function to install protoc with WORKDIR as argument
+install_protoc() {
+  # Determine architecture
+  if [[ $(uname -m) == "x86_64" ]]; then
+    ARCH="x86_64"
+  elif [[ $(uname -m) == "aarch64" ]] || [[ $(uname -m) == "arm64" ]] ; then
+    ARCH="aarch_64"
+  else
+    die "Unsupported architecture. Please consider manual installation."
+  fi
+  # Detect the Operating System
+  case "$(uname -s)" in
+    "Darwin") download_binary $ARCH "osx" "$1";;
+    "Linux") download_binary $ARCH "linux" $1;;
+  *) echo "Please consider manual installation from \
+     protocolbuffers/protobuf/releases and add to PATH" ;;
```

Comment:
"protocolbuffers/protobuf/releases" Does not make sense here. please provide the actual github URL: `https://github.com/protocolbuffers/protobuf/releases/`